### PR TITLE
Add support for PHP 8.2 to composer.json

### PR DESCRIPTION
**Pull request by TimWolla**
|    Q          |   A
|-------------- | ------
| Documentation | no
| Bugfix        | no
| BC Break      | no
| New Feature   | yes
| RFC           | no
| QA            | no

### Description

No code change needed, all tests passed locally. My understanding is that CI will automatically run tests for all versions in composer.json.

Furthermore my understanding of the last TSC meeting is that “security only” is redefined to “passive maintenance”, this change should qualify for that.

### CODE REVIEW

**Comment by TimWolla** [comment]
Okay, somehow even the older PHP versions broke compared to my previous #14. I'll investigate that.

File: composer.json [hunk 1]
```diff
@@ -19,7 +19,7 @@
         "sort-packages": true
     },
     "require": {
-        "php": "^7.3 || ~8.0.0 || ~8.1.0",
+        "php": "^7.3 || ~8.0.0 || ~8.1.0 || ~8.2.0",
```
Comment (Ocramius) [inline composer.json]: You can safely drop `^7.3` here, BTW

**Comment by Ocramius** [comment]
Please also run a `composer update nothing` (or equivalent) to re-generate the lockfile

**Comment by TimWolla** [comment]
Thanks for the review, both done. I'll look into the remaining issues later.

**Comment by Ocramius** [comment]
Hmm, I think we just need a newer PHPUnit version (and associated `phpunit.xml.dist`?)

**Comment by Ocramius** [comment]
Now needs a rebase as per #19.

**Comment by TimWolla** [comment]
Now "rebased" (rather: recreated). laminas-session needs a release first, laminas-json could be dropped, since we support only PHP greater than 8 now.

**Comment by Ocramius** [comment]
> laminas-session needs a release first

I'm watching the org activity, so send patches whenever you can, and I'll get them :ship: -ed and released ASAP.

 > laminas-json could be dropped

Agreed

File: src/Adapter/JsPull.php [hunk 2]
```diff
@@ -51,7 +51,7 @@ public function notify($current, $max, $percent, $timeTaken, $timeRemaining, $te
             'finished'      => false
         ];
 
-        $data = Json::encode($arguments);
+        $data = json_encode($arguments);
```
Comment (Ocramius) [inline src/Adapter/JsPull.php]: Important: `Json::encode()` and `Json::decode()` caused **exceptions** to be thrown, when using `laminas/laminas-json`.

We want to use `JSON_THROW_ON_ERROR` here.

File: src/Adapter/JsPush.php [hunk 3]
```diff
@@ -73,7 +73,7 @@ public function notify($current, $max, $percent, $timeTaken, $timeRemaining, $te
         ];
 
         $data = '<script type="text/javascript">'
-              . 'parent.' . $this->updateMethodName . '(' . Json::encode($arguments) . ');'
+              . 'parent.' . $this->updateMethodName . '(' . json_encode($arguments) . ');'
```
Comment (Ocramius) [inline src/Adapter/JsPush.php]: we need `JSON_THROW_ON_ERROR` here

**Review by Ocramius — CHANGES_REQUESTED** [review]
@TimWolla I think we should add the `.laminas-ci.json` with a platform requirement exclusion for PHP 8.2, and move this forward as-is, for now. 

It would make CI green, at least, and allow us a merge.

File: src/Adapter/JsPull.php (hunk 2 above)
Comment (TimWolla) [reply src/Adapter/JsPull.php]: `json_*en*code` should be infallible here, but I added the flag for good measure, as requested.

I guess that's why defaults matter :innocent:

Comment (Ocramius) [reply src/Adapter/JsPull.php]: cheeky :P

**Review by Ocramius — APPROVED** [review]
Thanks @TimWolla! :ship: 

Note:

 * auto-closer can just STFU
 * PHPCS pipeline requires PHPCS upgrade: not happening here

**Pull request merged**